static/js/pathfinder.js

```diff
@@ -127,62 +123,59 @@ function reloadReports(){
                 $('#vulnerabilityReport').append('<option value="'+r.id+'">'+r.name+'</option>');
             }
         });
-    }
-    restRequest('POST', {'index':'reports'}, updateData, '/plugin/pathfinder/api');
-}
-
-function openSource(source_id){
-    viewSection('sources', '/advanced/sources');
-    setTimeout(function(s){ $('#profile-source-name').val(s).change(); }, 1000, source_id);
+    }).catch((error) => {
+        toast('Error reloading Pathfinder reports.', false);
+        console.error(error);
+    });
 }
 
 function checkScanStatus(){
-    function updateData(data){
-        number_finished = Object.keys(data.finished).length
-        number_failed = Object.keys(data.errors).length
-        if (data.pending.length == 0){
+    apiV2('POST', '/plugin/pathfinder/api', {'index':'status'}).then((response) => {
+        number_finished = Object.keys(response.finished).length
+        number_failed = Object.keys(response.errors).length
+        if (response.pending.length == 0){
             validateFormState(true, '#startScan');
             if(number_finished == 0 && number_failed == 0){
                 clearInterval(refresher);
             }
         }
         if (number_finished > 0){
             source_id = '';
-            for (var key in data.finished){
-                displayOutput('scan ID:'+key+' finished. new source created: '+data.finished[key].source);
-                source_id = data.finished[key].source_id;
+            for (var key in response.finished){
+                displayOutput('scan ID:'+key+' finished. new source created: '+response.finished[key].source);
```

Comment:
This is nitpicky, but for string concatenation, templating tends to be a bit faster and JS linters tend to prefer them.

For example, this line would look like \`scan ID:${key} finished. new source created: ${response.finished[key].source}\`

Reply:
All `displayOutput` strings now use templating in the manner you suggested.  Thank you for pointing out this improvement.